hooks/fetch/use-API-route-data.ts

```diff
@@ -80,7 +80,15 @@ async function fetchAPIRoute<T extends APIRoutesPaths>(
     if (e instanceof FailToFetchError) {
       e.context.slug = slug;
       e.context.page = '/api/data-fetching/' + route;
-      if (!e.status || [408, 504, 429, 401].includes(e.status)) {
+      if (e.status === 429) {
+        logWarningInSentry(e);
+        if (e.message === 'Agent over rate limits') {
+          return IDataFetchingState.AGENT_OVER_RATE_LIMITS;
```

Comment:
I dont like the fact that this sound very bug prone

If one modify the message, this completly breaks